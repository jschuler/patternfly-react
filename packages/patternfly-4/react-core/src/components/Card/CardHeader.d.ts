import { FunctionComponent, HTMLProps, ReactType } from 'react';
import { TitleSize, TitleLevel } from '../Title';
import { OneOf, Omit } from '../../helpers/typeUtils';

export interface CardHeaderProps extends Omit<HTMLProps<HTMLHeadingElement>, 'size'> {
  component?: ReactType<CardHeaderProps>;
  size?: OneOf<typeof TitleSize, keyof typeof TitleSize>;
  headingLevel?: TitleLevel;
}

declare const CardHeader: FunctionComponent<CardHeaderProps>;

export default CardHeader;
