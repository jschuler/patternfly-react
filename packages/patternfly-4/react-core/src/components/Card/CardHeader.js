import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly/components/Card/card.css';
import { Title, TitleSize, TitleLevel } from '../Title';

const propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
  /** the size of the Title  */
  size: PropTypes.oneOf(Object.values(TitleSize)),
  /** the heading level to use */
  headingLevel: PropTypes.oneOf(Object.values(TitleLevel)),
  /** Additional props are spread to the container component */
  '': PropTypes.any
};

const defaultProps = {
  children: null,
  className: '',
  size: TitleSize.lg,
  headingLevel: TitleLevel.h1
};

const CardHeader = ({ children, className, size, headingLevel, ...props }) => (
  <Title className={css(styles.cardHeader, className)} size={size} headingLevel={headingLevel} {...props}>
    {children}
  </Title>
);

CardHeader.propTypes = propTypes;
CardHeader.defaultProps = defaultProps;

export default CardHeader;
