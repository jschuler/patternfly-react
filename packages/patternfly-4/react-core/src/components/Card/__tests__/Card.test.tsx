import * as React from 'react';
import { shallow } from 'enzyme';
import { Card } from '../Card';
// any missing imports can usually be resolved by adding them here
import {  } from '..';

it('Card should match snapshot (auto-generated)', () => {
  const view = shallow(
    <Card
      children={<>ReactNode</>}
			className={"''"}
			component={'article'}
			isHoverable={false}
			isCompact={false}
    />);
  expect(view).toMatchSnapshot();
});

test('renders with PatternFly Core styles', () => {
  const view = shallow(<Card />);
  expect(view).toMatchSnapshot();
});

test('className is added to the root element', () => {
  const view = shallow(<Card className="extra-class" />);
  expect(view.prop('className')).toMatchSnapshot();
});

test('extra props are spread to the root element', () => {
  const testId = 'card';
  const view = shallow(<Card data-testid={testId} />);
  expect(view.prop('data-testid')).toBe(testId);
});

test('allows passing in a string as the component', () => {
  const component = 'section';
  const view = shallow(<Card component={component} />);
  expect(view.type()).toBe(component);
});

test('allows passing in a React Component as the component', () => {
  const Component = () => <div>im a div</div>;
  const view = shallow(<Card component={Component} />);
  expect(view.type()).toBe(Component);
});

test('card with isHoverable applied ', () => {
  const view = shallow(<Card isHoverable />);
  expect(view).toMatchSnapshot();
});

test('card with isCompact applied ', () => {
  const view = shallow(<Card isCompact />);
  expect(view).toMatchSnapshot();
});
