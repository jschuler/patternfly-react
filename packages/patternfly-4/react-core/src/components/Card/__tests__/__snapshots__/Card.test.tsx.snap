// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot (auto-generated) 1`] = `
<article
  className="pf-c-card ''"
>
  ReactNode
</article>
`;

exports[`card with isCompact applied  1`] = `
<article
  className="pf-c-card pf-m-compact"
/>
`;

exports[`card with isHoverable applied  1`] = `
<article
  className="pf-c-card pf-m-hoverable"
/>
`;

exports[`className is added to the root element 1`] = `"pf-c-card extra-class"`;

exports[`renders with PatternFly Core styles 1`] = `
<article
  className="pf-c-card"
/>
`;
